unified-installer/astra-unified-installer.sh

```diff
@@ -2841,14 +2874,18 @@ if trident_will_be_installed_or_modified; then
         if components_include_acp; then
             # Enable ACP if needed (includes ACP upgrade)
             if ! acp_is_enabled; then
-                if config_acp_image_is_custom || prompt_user_yes_no "Would you like to enable Astra Control Provisioner?"; then
+                if config_acp_image_is_custom || prompt_user_yes_no "Would you like to enable ACP?"; then
+                    # create trident-acp secret
+                    kubectl create secret docker-registry "$IMAGE_PULL_SECRET" --docker-username="$ASTRA_ACCOUNT_ID" --docker-password="$ASTRA_API_TOKEN" -n trident --docker-server="$TRIDENT_ACP_IMAGE_REGISTRY" --dry-run=client -o yaml > "$__GENERATED_TRIDENT_ACP_SECRET_FILE"
```

Comment:
Since this is duplicated as is, might want to just put it in a function. You can then also add error handling to it, in the event creating the secret fails. For example, error handling might look something like this:
```
    local -r secret_yaml="$(kubectl create secret docker-registry "$IMAGE_PULL_SECRET" --docker-username="$ASTRA_ACCOUNT_ID" --docker-password="$ASTRA_API_TOKEN" -n trident --docker-server="$TRIDENT_ACP_IMAGE_REGISTRY" --dry-run=client -o yaml 2> "$__ERR_FILE"
    local -r captured_err="$(get_captured_err)"
    if [ -z "$secret_yaml" ] || [ -n "$captured_err" ]; then
        add_problem "Failed to generate secret for ACS registry '$TRIDENT_ACP_IMAGE_REGISTRY': $captured_err"
        return 1
    fi
```
- The ` 2> "$__ERR_FILE"` redirects the command's error messages (which goes through stderr, which is indicated by the `2`, while `>` is the redirection) into an error file we maintain in the script for exactly this purpose. 
- The next line, `local -r captured_err="$(get_captured_err)"`, simply gets the value we saved in that error file.
- We then check if the secret was actually generated or if we captured an error